### PR TITLE
Full CRL Functionality

### PR DESCRIPTION
**Pull request by wl2002**
Still need to attach CLI interface to functionality, and create api endpoint

### CODE REVIEW

**Comment by wl2002** [comment]
Also, I'm not sure what the standard expiry time for a CRL is so I tested using random value. Will update this when I find out.

**Comment by grittygrease** [comment]
The CI server is failing on Linux. Are you sure the path is correct for the testdata?

**Comment by grittygrease** [comment]
You should be able to set the expiration for a CRL via the CLI/API input.

**Comment by wl2002** [comment]
I'll add in the expiration input from the CLI/API next. I noticed it was failing on Linux but not on OS X, I wasn't sure why. The paths look good to me

File: crl/crl_test.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package crl
+
+import (
+	"testing"
+)
+
+const (
+	serverCertFile = "testdata/ca.pem"
+	serverKeyFile  = "testdata/ca-key.pem"
+	tryTwoCert     = "testData/caTwo.pem"
```
Comment (kisom) [inline crl/crl_test.go]: `testdata`, not `testData` (downcase the 'D').

File: crl/crl.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
+/*
+
+Package crl exposes Certificate Revocation List generation functionality
```
Comment (kisom) [inline crl/crl.go]: The normal style for CFSSL package docs is:

```
// Package crl exposes...
package crl
```

**Comment by wl2002** [comment]
I have to delete a lot of commented lines still, wanted to make sure was nothing broken beforehand. I believe the CLI portion (minus allowing setting expiry time) is complete, working on API now

**Comment by wl2002** [comment]
Now ready for code review

**Comment by kisom** [comment]
LGTM.

File: cli/gencrl/gencrl.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+)
+
+var gencrlUsageText = `cfssl gencrl -- generate a new Certificate Revocation List
+
+Usage of gencrl:
+        cfssl gencrl INPUTFILE CERT KEY TIME
+
+Arguments:
+        INPUTFILE:    			 Text file with one serial number per line, use '-' for reading text from stdin
+        CERT:		  			 The certificate that is signing this CRL, use '-' for reading text from stdin
```
Comment (lziest) [inline cli/gencrl/gencrl.go]: do you mind using space for alignment here? Thanks!

**Comment by wl2002** [comment]
Silly sublime! Fixed spacing :)

**Pull request merged**